server/config.go

```diff
@@ -17,56 +17,460 @@ package server
 import (
 	"fmt"
 
+	"github.com/dolthub/dolt/go/cmd/dolt/commands/engine"
 	"github.com/dolthub/dolt/go/cmd/dolt/commands/sqlserver"
-	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/cluster"
 	"github.com/dolthub/dolt/go/libraries/utils/filesys"
 	"gopkg.in/yaml.v2"
 )
 
-type DoltgresServerConfig struct {
-	*sqlserver.YAMLConfig
-	PostgresReplicationConfig *PostgresReplicationConfig `yaml:"postgres_replication,omitempty"`
+func Ptr[T any](v T) *T {
+	return &v
 }
 
+const (
+	maxConnectionsKey = "max_connections"
+	readTimeoutKey    = "net_read_timeout"
+	writeTimeoutKey   = "net_write_timeout"
+	eventSchedulerKey = "event_scheduler"
+
+	OverrideDataDirKey = "data_dir"
+)
+
+var ConfigHelp = `Place holder. This will be replaced with details on config.yaml options`
+
 type PostgresReplicationConfig struct {
-	PostgresServerAddress string `yaml:"postgres_server_address"`
-	PostgresUser          string `yaml:"postgres_user"`
-	PostgresPassword      string `yaml:"postgres_password"`
-	PostgresDatabase      string `yaml:"postgres_database"`
-	PostgresPort          int    `yaml:"postgres_port"`
-	SlotName              string `yaml:"slot_name"`
+	PostgresServerAddress *string `yaml:"postgres_server_address,omitempty" minver:"TBD"`
+	PostgresUser          *string `yaml:"postgres_user,omitempty" minver:"TBD"`
+	PostgresPassword      *string `yaml:"postgres_password,omitempty" minver:"TBD"`
+	PostgresDatabase      *string `yaml:"postgres_database,omitempty" minver:"TBD"`
+	PostgresPort          *int    `yaml:"postgres_port,omitempty" minver:"TBD"`
+	SlotName              *string `yaml:"slot_name,omitempty" minver:"TBD"`
 }
 
-var _ sqlserver.ServerConfig = (*DoltgresServerConfig)(nil)
+// BehaviorYAMLConfig contains server configuration regarding how the server should behave
+type DoltgresBehaviorConfig struct {
+	ReadOnly   *bool `yaml:"read_only,omitempty" minver:"TBD"`
+	AutoCommit *bool `yaml:"autocommit,omitempty" minver:"TBD"`
+	// PersistenceBehavior regulates loading persisted system variable configuration.
+	PersistenceBehavior *string `yaml:"persistence_behavior,omitempty" minver:"TBD"`
```

Comment:
I think we want to kill this as well, make it always true

Reply:
true? for PersistenceBehavior is a string.  Default dolt behavior is "load".